cmd/endpoint.go

```diff
@@ -228,25 +229,61 @@ func localEndpointsMemUsage(endpoints EndpointList) ServerMemUsageInfo {
 	}
 }
 
+// localEndpointsNetStats - returns ServerNetStatsInfo for only the
+// local endpoints from given list of endpoints
+func localEndpointsNetStats(endpoints EndpointList) ServerNetStatsInfo {
+	var tputs []xnet.Stats
+	var oneMinTputs []xnet.Stats
+	var fiveMinTputs []xnet.Stats
+	var fifteenMinTputs []xnet.Stats
+	var addr string
+	seenEndpoints := set.NewStringSet()
+	for _, endpoint := range endpoints {
+		// Only proceed for local endpoints
+		if endpoint.IsLocal {
+			if seenEndpoints.Contains(endpoint.Host) {
+				continue
+			}
+			addr = GetLocalPeer(endpoints)
```

Comment:
You don't need to do this for all local endpoints since `addr` is only one value this can be outside the for loop i.e before the for loop

```
addr := GetLocalPeer(endpoints)
```

Automatically gives you the local addr - since it internally validates endpoints.IsLocal()